CustomRules.rb

```diff
@@ -0,0 +1,72 @@
+$dict = {3 => 'Fizz', 5 => 'Buzz', 7 => 'Bang', 11 => 'Bong', 13 => 'Fezz'}
+
+def rule_3(number, output)
+  if number % 3 == 0
+    output << $dict[3]
+  end
+  output
+end
+
+def rule_5(number,output)
+  if number % 5 == 0
+    output << $dict[5]
+  end
+  output
+end
+
+def rule_7(number,output)
+  if number % 7 == 0 && output != []
+    output << $dict[7]
+  end
+  output
+end
+
+def rule_11(number,output)
+  if number % 11 == 0
+    output = $dict[11]
+  end
+  output
+end
+
+def rule_13(number,output)
+  if number % 13 == 0
+    index = output.index {|element| element[0] == 'B'}
+    if index.nil?
+      output << $dict[13]
+    else
+      output.insert(index,$dict[13])
+    end
+  end
+  output
+end
+
+def rule_17(number,output)
+  if number % 17 == 0
+    output.reverse!
+  end
+  output
+end
+
+def custom_rules(rules,n)
+  for number in 1..n
+
+    output = []
+    for rule_number in rules
+      function_name = "rule_#{rule_number}"
+      output = send(function_name, number, output)

```

Comment:
Nice! Dynamically calling a different method based on input like this is really cool and makes for very neat code. As you've written it though it's not very robust. A user could enter `12` as a rule and then this would give them an error. Any ideas what you could do about that?

Reply:
You'll also do the rules in the order the user provides them. Is that going to cause problems?

Reply:
Thank you for the comments! I have added some changes to account for these issues:
- Sort the rules before execution.
- Add a validation_input function which checks that the rules from the input are indeed from the rules list.

Reply:
Fantastic, looks great.